### PR TITLE
config: drop `skip_plume_release_task` hack, add `misc.generate_release_index`

### PR DESCRIPTION
**Pull request by jlebon**
Rather than `plume release`, use the new `plume make-amis-public` and
`plume update-release-index` commands directly. Gate the former on
whether the AMIs were already made public on first upload. Gate the
latter on a new `misc.generate_release_index` since we only need it for
FCOS. Finally, get rid of the `skip_plume_release_task` knob.

### CODE REVIEW

**Comment by jlebon** [comment]
Requires: https://github.com/coreos/coreos-assembler/pull/3240

File: jobs/release.Jenkinsfile [hunk 1]
```diff
@@ -298,19 +298,25 @@ lock(resource: "release-${params.STREAM}", extra: locks) {
...
-                    // We'll make this more explicit in the future.
+                def bucket_prefix = "${pipecfg.s3.bucket}/${pipecfg.s3.builds_key}"
+
+                // make AMIs public if not already the case
+                if (!pipecfg.clouds?.aws?.public) {
+                    shwrap("""
+                    cosa shell -- plume make-amis-public \
+                        --version ${params.VERSION} \
+                        --stream ${params.STREAM} \
+                        --bucket-prefix ${bucket_prefix} \
```
Comment (dustymabe) [inline jobs/release.Jenkinsfile:309]: I still need to review the COSA PR, but just from scrolling this it seems odd that a call to make AMIs public needs info about an s3 bucket.

Comment (jlebon) [reply jobs/release.Jenkinsfile:309]: It reads the release metadata directly from the bucket. `plume` commands traditionally are not "cosa workdir-aware", so I'd prefer not to have it start looking for a `builds/` dir there. But what we could do is hand it the release metadata file directly ourselves. That should actually allow us to drop the `--version` and `--stream` switches too.

I'll see about enhancing it for that, but will do that as a follow-up to https://github.com/coreos/coreos-assembler/pull/3240 so that we can get rid of `plume release` first to not support the old way anymore.

File: docs/config.yaml [hunk 2]
```diff
@@ -110,6 +108,9 @@ s3:
   # OPTIONAL: object ACL to use when uploading
   acl: private
 
+# OPTIONAL: whether to generate a release index
+release_index: true
```
Comment (dustymabe) [inline docs/config.yaml:112]: maybe we can add this under a new `misc` section and collect other loose items under that? I had started to do this in https://github.com/coreos/fedora-coreos-pipeline/pull/759 (see [this](https://github.com/coreos/fedora-coreos-pipeline/compare/35cfa7dc3cd8c7052eedf7eb27079b654f60c1bb..cb1cfa71d8cc56f8fba0fe79c87d8a451f0937e6) force push), but we ended up going a different way.

TL;DR - I think things are starting to get a bit disorganized and having a misc "things that don't fit anywhere else" section might be nice.

File: jobs/release.Jenkinsfile [hunk 3]
```diff
@@ -298,19 +298,25 @@ lock(resource: "release-${params.STREAM}", extra: locks) {
...
+                        --aws-credentials \${AWS_BUILD_UPLOAD_CONFIG}
```
Comment (dustymabe) [inline jobs/release.Jenkinsfile:310]: do we need to consider govcloud here?

Comment (jlebon) [reply jobs/release.Jenkinsfile:310]: Yeah, good point. I guess we will have to enhance `make-amis-public` to also take a GovCloud credentials and use it when iterating on a GovCloud region. It's not immediately necessary since we already make the AMIs public at upload so we won't fall in this branch, but if that ever changes we would. Will track that in a cosa issue so we don't forget it.

**Comment by jlebon** [comment]
Updated!

**Review by dustymabe — APPROVED** [review]
LGTM

File: jobs/release.Jenkinsfile (hunk 3 above)
Comment (jlebon) [reply jobs/release.Jenkinsfile:310]: Opened https://github.com/coreos/coreos-assembler/issues/3259.
